azurerm_monitor_scheduled_query_rules_alert.action optional custom webhook required

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform = 0.12.20
AzureRM = 2.3.0

### Affected Resource(s)

`azurerm_monitor_scheduled_query_rules_alert`

### Terraform Configuration Files


```hcl
resource "azurerm_monitor_scheduled_query_rules_alert" "example" {
  name                = "example"
  location            = var.location
  resource_group_name = var.logging_resource_group

  action {
    action_group           = var.action_group_ids
  }
```

### Expected Behavior

Terraform should apply with the custom webhook setting disabled for the query rule. 

### Actual Behavior

Terraform tries to send a blank custom payload = "" resulting in the following error :
```hcl
##[error]Error: Error creating or updating Scheduled Query Rule "example" (resource group "example"): insights.ScheduledQueryRulesClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: error response cannot be parsed: "{\r\n  \"error\": {\r\n    \"code\": \"Custom Webhook Payload is not valid: Error reading JObject from JsonReader. Path '', line 0, position 0.\",\r\n    \"message\": {\r\n      \"lang\": \"en-US\",\r\n      \"value\": \"Custom Webhook Payload is not valid: Error reading JObject from JsonReader. Path '', line 0, position 0.\"\r\n    }\r\n  }\r\n}" error: json: cannot unmarshal object into Go struct field RequestError.error of type string
```

### Steps to Reproduce

1. Create a scheduled query rule resource with an action block and no custom webhook payload defined. 
2. Terraform Apply
3. View JSON format error.

--- a/azurerm/internal/services/monitor/monitor_scheduled_query_rules_alert_resource.go
+++ b/azurerm/internal/services/monitor/monitor_scheduled_query_rules_alert_resource.go
@@ -75,7 +75,9 @@ func resourceArmMonitorScheduledQueryRulesAlert() *schema.Resource {
 						"custom_webhook_payload": {
 							Type:         schema.TypeString,
 							Optional:     true,
-							Default:      "{}",
+							// TODO remove `Computed: true` in 3.0. This is a breaking change where the Default used to be "{}"
+							// We'll keep Computed: true for users who expect the same functionality but will remove it in 3.0
+							Computed:     true,
 							ValidateFunc: validation.StringIsJSON,
 						},
 						"email_subject": {
@@ -418,7 +420,9 @@ func expandMonitorScheduledQueryRulesAlertAction(input []interface{}) *insights.
 		actionGroups := v["action_group"].(*schema.Set).List()
 		result.ActionGroup = utils.ExpandStringSlice(actionGroups)
 		result.EmailSubject = utils.String(v["email_subject"].(string))
-		result.CustomWebhookPayload = utils.String(v["custom_webhook_payload"].(string))
+		if v := v["custom_webhook_payload"].(string); v != "" {
+			result.CustomWebhookPayload = utils.String(v)
+		}
 	}
 
 	return &result

--- a/azurerm/internal/services/monitor/monitor_scheduled_query_rules_alert_resource.go
+++ b/azurerm/internal/services/monitor/monitor_scheduled_query_rules_alert_resource.go
@@ -73,8 +73,8 @@ func resourceArmMonitorScheduledQueryRulesAlert() *schema.Resource {
 							},
 						},
 						"custom_webhook_payload": {
-							Type:         schema.TypeString,
-							Optional:     true,
+							Type:     schema.TypeString,
+							Optional: true,
 							// TODO remove `Computed: true` in 3.0. This is a breaking change where the Default used to be "{}"
 							// We'll keep Computed: true for users who expect the same functionality but will remove it in 3.0
 							Computed:     true,

